### PR TITLE
Release version 5.0.0

### PR DESCRIPTION
**Pull request by aoune-ayoub**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

Description
New release 5.0.0 for OPTIGA Trust M host library. Changes include : 
* New folder structure.
* Mbedtls is now configured to be used a submodule from https://github.com/Mbed-TLS/mbedtls/tree/v2.28.7.
* Added support for Zehpyr OS.
* README.md update.
* Fix Metadata configuration for OID 0xE0E8.
* fix buffer length when reading PBS for shielded I2C connection.

Related Issue
New version Release.

### CODE REVIEW

File: include/optiga_lib_config_m_v3.h
Comment (RaymWong) [inline include/optiga_lib_config_m_v3.h:1]: Changing of core directory structures will break many other tools and customer codes in the field. Besides All internal tools need to be rechecked.

**Review by RaymWong — COMMENTED** [review]
Changing the core library structures especially the platform dependent pal layers might cause other issues, including customers codes in the field. For all internal tools will need to be modified accordingly.

File: include/optiga_lib_config_m_v3.h
Comment (aoune-ayoub) [reply include/optiga_lib_config_m_v3.h:1]: Issue was addressed.
Current version of Host library will not be affected and will stay active.
New version will not affect existent projects, examples and applications.
New projects will be ported to Release 5.0.0

**Review by spaehn — APPROVED** [review]
Looks good to me! Thanks for preparing this new release!

**Review by koblibri — APPROVED** [review]
Approved as discussed internally

**Pull request merged**